flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/utils/ExecutorUtils.java

```diff
@@ -70,6 +71,7 @@ public static void setBatchProperties(StreamGraph streamGraph, TableConfig table
         streamGraph.setScheduleMode(ScheduleMode.LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST);
         streamGraph.setStateBackend(null);
         streamGraph.setCheckpointStorage(null);
+        streamGraph.setJobType(JobType.BATCH);
```

Comment:
let's move this right next to the schedule mode, and add a comment that this is dependent on the schedule mode.